ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -631,18 +633,14 @@ void fillChunks() {
         }
       }
       if (operationCompleted) {
-        PutChunk lastChunk = getBuildingChunk();
-        if (lastChunk != null) {
-          logger.info("{}: Clear unfinished chunk since operation is completed", loggingContext);
-          // call release blob content, not clear, since clear should only be used in the main thread.
-          lastChunk.releaseBlobContent();
```

Comment:
Was this removed on purpose?

Reply:
this is not removed, it's merged in the for loop block below. In the for loop we are clearing the building chunk as well.